### PR TITLE
Add support for virtual bases in GetBaseClassOffset

### PR DESCRIPTION
**Pull request by sudo-panda**
Fixes Base class not found error.

### CODE REVIEW

File: unittests/InterOp/ScopeReflectionTest.cpp [hunk 1]
```diff
@@ -544,6 +544,7 @@ TEST(ScopeReflectionTest, GetBaseClassOffset) {
     class B { int b; };
     class C : public A, public B { int c; };
     class D : public A, public B, public C { int d; };
+    class E : virtual A { int e; };
```
Comment (vgvassilev) [inline unittests/InterOp/ScopeReflectionTest.cpp:547]: Our test should include at least two base classes with a common base. Then in the check we should ensure that there is only one base.

Comment (vgvassilev) [reply unittests/InterOp/ScopeReflectionTest.cpp:547]: Perhaps the case `D` above is better to check.

File: unittests/InterOp/ScopeReflectionTest.cpp [hunk 2]
```diff
@@ -554,6 +555,7 @@ TEST(ScopeReflectionTest, GetBaseClassOffset) {
   EXPECT_EQ(InterOp::GetBaseClassOffset(S, Decls[3], Decls[0]), 0);
   EXPECT_EQ(InterOp::GetBaseClassOffset(S, Decls[3], Decls[1]), 4);
   EXPECT_EQ(InterOp::GetBaseClassOffset(S, Decls[3], Decls[2]), 8);
+  EXPECT_EQ(InterOp::GetBaseClassOffset(S, Decls[4], Decls[0]), 12);
```
Comment (vgvassilev) [inline unittests/InterOp/ScopeReflectionTest.cpp:558]: Why here it is 12?

File: unittests/InterOp/ScopeReflectionTest.cpp (hunk 1 above)
Comment (sudo-panda) [reply unittests/InterOp/ScopeReflectionTest.cpp:547]:
```c++
class F : virtual A { int f; }
class G : public F, public E { int g; }
```
Like this?

File: unittests/InterOp/ScopeReflectionTest.cpp (hunk 2 above)
Comment (sudo-panda) [reply unittests/InterOp/ScopeReflectionTest.cpp:558]: https://godbolt.org/z/3db7M9sM6

I don't know why but it checks out

Comment (vgvassilev) [reply unittests/InterOp/ScopeReflectionTest.cpp:558]: Ok, fair enough...

**Comment by vgvassilev** [comment]
This is quite nicer! Many thanks!

**Pull request merged**